python-lib/dku_timeseries/stl_decomposition.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+from statsmodels.tsa.seasonal import STL
+
+from dku_timeseries.decomposition import TimeseriesDecomposition
+
+
+class STLDecomposition(TimeseriesDecomposition):
+    def __init__(self, dku_config):
+        super().__init__(dku_config)
+        self.parameters = format_parameters(dku_config)
+
+    def _decompose(self, ts):
+        if self.dku_config.model == "multiplicative":
+            self.parameters["endog"] = np.log(ts)
+            stl = STL(**self.parameters)
+            results = stl.fit()
+            results._trend = np.exp(results.trend)
```

Comment:
do you re-use the `results._trend` (same for `_seasonal` and `_resid`) somewhere ?
in the `_write_decomposition()` function I only see `results.trend`

Reply:
It refers to the object `DecomposeResult `, defined in [Statsmodel](https://github.com/statsmodels/statsmodels/blob/master/statsmodels/tsa/seasonal.py) .`trend`is a property which returns the attribute `result._trend`

Reply:
I don't think I can change the trend property directly

Reply:
Thanks for the link, it makes it clearer.
Modifying a class variable starting with "_" (meaning it's supposed to be only for internal use) is not a great practice.
To avoid that, maybe you could also implement the `_write_decomposition` method in the child class `STLDecomposition` to handle the case with `np.exp`

Reply:
I added an inner class `_DecompositionResults` to manage these values without modifying the inner attributes of Statsmodel `DecomposeResults`in https://github.com/dataiku/dss-plugin-timeseries-preparation/pull/13/commits/fda53d7d0d46733f51895d1218d7fb1b7bb15984

Reply:
actually, I'm not sure wether it's good practice to use an inner class this way, maybe @MayeulRousselet , could we talk about this together?

Reply:
I don't think you should use an inner class, move the class outside (but in the same module).

Reply:
done in https://github.com/dataiku/dss-plugin-timeseries-preparation/pull/13/commits/77420bfe240d7982b82d53ae1bba8bbc3b22c168